ot/lowrank.py

```diff
@@ -8,14 +8,138 @@
 
 
 import warnings
-from .utils import unif, get_lowrank_lazytensor
+from .utils import unif, dist, get_lowrank_lazytensor
 from .backend import get_backend
+from .bregman import sinkhorn
+from sklearn.cluster import KMeans
 
 
-def compute_lr_sqeuclidean_matrix(X_s, X_t, nx=None):
+def _init_lr_sinkhorn(X_s, X_t, a, b, rank, init, reg_init=None, random_state=None, nx=None):
+    """
+    Implementation of different initialization strategies for the low rank sinkhorn solver (Q ,R, g).
+    This function is specific to lowrank_sinkhorn.
+
+    Parameters
+    ----------
+    X_s : array-like, shape (n_samples_a, dim)
+        samples in the source domain
+    X_t : array-like, shape (n_samples_b, dim)
+        samples in the target domain
+    a : array-like, shape (n_samples_a,)
+        samples weights in the source domain
+    b : array-like, shape (n_samples_b,)
+        samples weights in the target domain
+    rank : int
+        Nonnegative rank of the OT plan.
+    init : str
+        Initialization strategy for Q, R and g. 'random', 'trivial' or 'kmeans'
+    reg_init : float, optional. Default is None. (>0)
+        Regularization term for a 'kmeans' init. If None, 1 is considered.
+    random_state : default None
+        Random state for a "random" or 'kmeans' init strategy
+    nx : default None
+        POT backend
+
+
+    Returns
+    ---------
+    Q : array-like, shape (n_samples_a, r)
+        Init for the first low-rank matrix decomposition of the OT plan (Q)
+    R: array-like, shape (n_samples_b, r)
+        Init for the second low-rank matrix decomposition of the OT plan (R)
+    g : array-like, shape (r, )
+        Init for the weight vector of the low-rank decomposition of the OT plan (g)
+
+
+    References
+    -----------
+    .. [65] Scetbon, M., Cuturi, M., & Peyré, G. (2021).
+        "Low-rank Sinkhorn factorization". In International Conference on Machine Learning.
+
+    """
+
+    if nx is None:
+        nx = get_backend(X_s, X_t, a, b)
+
+    if reg_init is None:
```

Comment:
You can also directly set it to `reg_init=1.`.
The logic is that it does not dependent explicitly on other inputs of the function so there is no need to add a layer of abstraction via `None` default value.